process/block/shardinfo_test.go

```diff
@@ -0,0 +1,1098 @@
+			return true
+		}
+
+		sic, err := NewShardInfoCreateData(
+			args.enableEpochsHandler,
+			pool.Headers(),
+			args.proofsPool,
+			args.pendingMiniBlocksHandler,
+			args.blockTracker,
+		)
+		require.Nil(t, err)
+		shardInfo, err := sic.CreateShardInfoV3(metaHdrV3, headers, headerHashes)
+		require.Nil(t, err)
+		require.NotNil(t, shardInfo)
+		require.Equal(t, 4, len(shardInfo))
+	})
+}
+
+func TestShardInfoCreateData_CreateShardInfoFromLegacyMeta(t *testing.T) {
+	t.Parallel()
+
+	pool := dataRetrieverMock.NewPoolsHolderMock()
+	// we will have a 3 hdrs in pool
+	hdrHash1 := []byte("hdr hash 1")
+	hdrHash2 := []byte("hdr hash 2")
+	hdrHash3 := []byte("hdr hash 3")
+
+	mbHash1 := []byte("mb hash 1")
+	mbHash2 := []byte("mb hash 2")
+	mbHash3 := []byte("mb hash 3")
+
+	miniBlockHeader1 := block.MiniBlockHeader{Hash: mbHash1}
+	miniBlockHeader2 := block.MiniBlockHeader{Hash: mbHash2}
+	miniBlockHeader3 := block.MiniBlockHeader{Hash: mbHash3}
+
+	miniBlockHeaders1 := make([]block.MiniBlockHeader, 0)
+	miniBlockHeaders1 = append(miniBlockHeaders1, miniBlockHeader1)
+	miniBlockHeaders1 = append(miniBlockHeaders1, miniBlockHeader2)
+	miniBlockHeaders1 = append(miniBlockHeaders1, miniBlockHeader3)
+
+	miniBlockHeaders2 := make([]block.MiniBlockHeader, 0)
+	miniBlockHeaders2 = append(miniBlockHeaders2, miniBlockHeader1)
+	miniBlockHeaders2 = append(miniBlockHeaders2, miniBlockHeader2)
+
+	miniBlockHeaders3 := make([]block.MiniBlockHeader, 0)
+	miniBlockHeaders3 = append(miniBlockHeaders3, miniBlockHeader1)
+
+	header1 := &block.Header{
+		Round:            1,
+		Nonce:            45,
+		ShardID:          0,
+		MiniBlockHeaders: miniBlockHeaders1}
+	header2 := &block.Header{
+		Round:            2,
+		Nonce:            45,
+		ShardID:          1,
+		MiniBlockHeaders: miniBlockHeaders2}
+	header3 := &block.Header{
+		Round:            3,
+		Nonce:            45,
+		ShardID:          2,
+		MiniBlockHeaders: miniBlockHeaders3}
+	// put the existing headers inside datapool
+	pool.Headers().AddHeader(hdrHash1, header1)
+	pool.Headers().AddHeader(hdrHash2, header2)
+	pool.Headers().AddHeader(hdrHash3, header3)
+
+	headerHashes := make([][]byte, 0)
+	headers := make([]data.ShardHeaderHandler, 0)
+	headers = append(headers, []data.ShardHeaderHandler{header1, header2, header3}...)
+	headerHashes = append(headerHashes, [][]byte{hdrHash1, hdrHash2, hdrHash3}...)
+	round := uint64(10)
+	metaHdr := &block.MetaBlock{Round: round}
+
+	t.Run("should fail with V3 meta header", func(t *testing.T) {
+		t.Parallel()
+
+		args := createDefaultShardInfoCreateDataArgs()
+		metaHdrV3 := &block.MetaBlockV3{Round: round}
+		sic, err := NewShardInfoCreateData(
+			args.enableEpochsHandler,
+			args.headersPool,
+			args.proofsPool,
+			args.pendingMiniBlocksHandler,
+			args.blockTracker,
+		)
+		require.Nil(t, err)
+		shardInfo, err := sic.CreateShardInfoFromLegacyMeta(metaHdrV3, headers, headerHashes)
+		require.NotNil(t, err)
```

Comment:
can you test that the error is the expected one, not just that there is an error?